lib/pure/ospaths.nim

```diff
@@ -548,23 +548,21 @@ proc getTempDir*(): string {.rtl, extern: "nos$1",
 
 proc expandTilde*(path: string): string {.
   tags: [ReadEnvEffect, ReadIOEffect].} =
-  ## Expands a path starting with ``~/`` to a full path.
+  ## Expands ``~`` or a path starting with ``~/`` to a full path, replacing
+  ## ``~`` with ``getHomeDir()`` (otherwise returns ``path`` unmodified).
   ##
-  ## If `path` starts with the tilde character and is followed by `/` or `\\`
-  ## this proc will return the reminder of the path appended to the result of
-  ## the getHomeDir() proc, otherwise the input path will be returned without
-  ## modification.
-  ##
-  ## The behaviour of this proc is the same on the Windows platform despite
-  ## not having this convention. Example:
-  ##
-  ## .. code-block:: nim
-  ##   let configFile = expandTilde("~" / "appname.cfg")
-  ##   echo configFile
-  ##   # --> C:\Users\amber\appname.cfg
-  if len(path) > 1 and path[0] == '~' and (path[1] == '/' or path[1] == '\\'):
```

Comment:
Why not just:

```nim
if len(path) >= 1 and path[0] == '~' and (path.len == 1 or path[1] in {DirSep, AltSep}):
```

or if you like separate if branches then I would prefer this way:

```nim
if path.len >= 1 and path[0] == '~':
  return getHomeDir() / path.substr(2)
else:
  return path
```

Reply:
your suggestion returns wrong results:
```nim
echo test("~") # returns /home/timothee/ instead of /home/timothee
echo test("~bob") # returns /home/timothee/ob instead of `~bob` (unmodified, cf TODO I left for future PR) or the more correct /home/bob (future PR)
```

Reply:
Why is a trailing slash such a big issue?

In any case, I'm sure you can adjust my code to fix these. I wrote these quickly, the main point is the style of code. The ordering of your if branches in this PR is a little confusing.